pennylane/templates/state_preparations/qrom_state_prep.py

```diff
@@ -0,0 +1,281 @@
+            ("precision_wires", self.hyperparameters["precision_wires"]),
+            ("work_wires", self.hyperparameters["work_wires"]),
+        )
+        return (self.state_vector,), hyperparameters
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        hyperparams_dict = dict(metadata)
+        return cls(data[0], **hyperparams_dict)
+
+    def map_wires(self, wire_map):
+        new_wires = Wires(
+            [wire_map.get(wire, wire) for wire in self.hyperparameters["input_wires"]]
+        )
+        new_precision_wires = Wires(
+            [wire_map.get(wire, wire) for wire in self.hyperparameters["precision_wires"]]
+        )
+
+        new_work_wires = Wires(
+            [wire_map.get(wire, wire) for wire in self.hyperparameters["work_wires"]]
+        )
+        return QROMStatePreparation(
+            self.state_vector, new_wires, new_precision_wires, new_work_wires
+        )
+
+    @staticmethod
+    def compute_decomposition(
+        state_vector, wires, input_wires, precision_wires, work_wires
+    ):  # pylint: disable=arguments-differ
+        r"""
+        Computes the decomposition operations for the given state vector.
+
+        Args:
+
+            state_vector (TensorLike): The state vector to prepare.
+            wires (Sequence[int]): The wires which the operator acts on.
+            input_wires (Sequence[int]): The wires on which to prepare the state.
+            precision_wires (Sequence[int]): The wires allocated for storing the binary representations of the
+                rotation angles utilized in the template.
+            work_wires (Sequence[int]):  The wires used as work wires for the QROM operations. Defaults to ``None``.
+
+        Returns:
+            list: List of decomposition operations.
+        """
+
+        probs = qml.math.abs(state_vector) ** 2
+        phases = qml.math.angle(state_vector) % (2 * np.pi)
+        eps = 1e-15  # Small constant to avoid division by zero
+
+        decomp_ops = []
+        num_iterations = int(qml.math.log2(qml.math.shape(probs)[0]))
+        rotation_angles = [2 ** (-ind - 1) for ind in range(len(precision_wires))]
+
+        for i in range(num_iterations):
+
+            probs_aux = qml.math.reshape(probs, [1, -1])
+
+            # Calculation of the numerator and denominator of the function f(x) (Eq.5 [arXiv:quant-ph/0208112])
+            for itx in range(i + 1):
+                probs_denominator = qml.math.sum(probs_aux, axis=1)
+                probs_aux = qml.math.reshape(probs_aux, [int(2 ** (itx + 1)), -1])
+                probs_numerator = qml.math.sum(probs_aux, axis=1)[::2]
+
+            # Compute the binary representations of the angles θi
+            thetas_binary = [
+                _float_to_binary(
+                    2
+                    / np.pi
+                    * qml.math.arccos(
+                        qml.math.sqrt(probs_numerator[j] / (probs_denominator[j] + eps))
+                    ),
```

Comment:
```suggestion
                    2 * qml.math.arccos(
                        qml.math.sqrt(probs_numerator[j] / (probs_denominator[j] + eps))
                    ) / np.pi,
```
does this result in better formatting?